exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggBatch.java

```diff
@@ -255,7 +254,6 @@ private HashAggregator createAggregatorInternal() throws SchemaChangeException,
       groupByOutFieldIds[i] = container.add(vv);
     }
 
-    int extraNonNullColumns = 0; // each of SUM, MAX and MIN gets an extra bigint column
```

Comment:
Thanks for catching this. Then we should fix the underlying problem instead of passing around additional parameters to work around the issue. I will work on fixing the codegen for the BatchHolder as part of this PR.

Reply:
Maybe do this work as a separate PR (for DRILL-5728) ?  Else it would delay this PR, and overload it ...

Reply:
Ok. I am going to move all the code into one place though. Previously half the code was in HashAggBatch and the other half was in HashAggTemplate. I'll also add a comment that this extra code should be removed after DRILL-5728, and I will also add a comment to DRILL-5728 to reminder the implementer to remove this extra code.